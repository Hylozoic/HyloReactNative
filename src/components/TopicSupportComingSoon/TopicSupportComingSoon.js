import React from 'react'
import { View, Text, Image, TouchableOpacity } from 'react-native'
import styles from './TopicSupportComingSoon.styles'

const axolotlImage = require('../../assets/Axolotyl_Digging.png')

export default function TopicSupportComingSoon ({ style, navigation }) {
  return <View style={[style, styles.container]}>
    <View style={styles.header}>
      <Text style={styles.headerText}>
        We're working on expanding #topics to more places.
      </Text>
    </View>
    <View style={styles.paddedRow}>
      <Text style={styles.bodyText}>
        In the meantime, click a #topic from an individual
        community to see posts from that community.
      </Text>
    </View>
    <Image style={styles.image} resizeMode='stretch' source={axolotlImage} />
    <View style={styles.paddedRow}>
      <TouchableOpacity onPress={() => navigation.goBack()} style={styles.gotBackButton}>
        <Text style={styles.gotBackButtonText}>Go Back</Text>
      </TouchableOpacity>
    </View>
  </View>
}
