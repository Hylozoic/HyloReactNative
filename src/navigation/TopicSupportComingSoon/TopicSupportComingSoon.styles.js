import {
  capeCod,
  rhino60,
  caribbeanGreen
} from 'style/colors'

export default {
  container: {
    backgroundColor: 'white',
    flex: 1,
    alignItems: 'center',
    flexDirection: 'column'
  },
  header: {
    marginTop: 25
  },
  headerText: {
    color: capeCod,
    fontSize: 24,
    fontFamily: 'Circular-Bold',
    textAlign: 'center',
    marginBottom: 25
  },
  bodyText: {
    color: rhino60,
    fontSize: 16,
    fontFamily: 'Circular-Bold',
    textAlign: 'center'
  },
  image: {
    flex: 1
  },
  goBackButton: {
    flexDirection: 'row',
    backgroundColor: caribbeanGreen,
    height: 36,
    width: 200,
    borderRadius: 50,
    justifyContent: 'center'
  },
  goBackButtonText: {
    color: 'white',
    backgroundColor: 'transparent',
    textAlign: 'center',
    fontFamily: 'Circular-Book',
    fontSize: 18,
    lineHeight: 32
  },
  paddedRow: {
    paddingRight: 15,
    paddingLeft: 15,
    marginBottom: 15
  }
}
