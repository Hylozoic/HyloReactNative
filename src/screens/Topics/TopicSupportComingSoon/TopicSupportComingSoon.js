import React from 'react'
import { View, Text, Image } from 'react-native'
import SafeAreaView from 'react-native-safe-area-view'
import LinkButton from 'navigation/linking/LinkButton'
import styles from './TopicSupportComingSoon.styles'

const axolotlImage = require('assets/Axolotyl_Digging.png')

export default function TopicSupportComingSoon ({ style, navigation }) {
  return (
    <SafeAreaView style={[style, styles.container]}>
      <View style={styles.header}>
        <Text style={styles.headerText}>
          We're working on expanding #topics to more places.
        </Text>
      </View>
      <View style={styles.paddedRow}>
        <Text style={styles.bodyText}>
          In the meantime, click a #topic from an individual community to see posts from that community.
        </Text>
      </View>
      <Image style={styles.image} resizeMode='contain' source={axolotlImage} />
      <View style={styles.paddedRow}>
        <LinkButton to='/' style={styles.goBackButton}>
          <Text style={styles.goBackButtonText}>Go Back</Text>
        </LinkButton>
      </View>
    </SafeAreaView>
  )
}
