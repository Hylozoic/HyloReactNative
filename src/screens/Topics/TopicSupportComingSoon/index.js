import component from './TopicSupportComingSoon'

export default component
