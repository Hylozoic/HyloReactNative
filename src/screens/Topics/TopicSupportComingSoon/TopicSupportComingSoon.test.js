import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import TopicSupportComingSoon from './TopicSupportComingSoon'

it('matches last snapshot', () => {
  const renderer = new ReactShallowRenderer()
  const styles = {
    container: {
      some: 'styles'
    }
  }
  const route = {
    params: {
      onBack: jest.fn()
    }
  }
  const navigation = {
    goBack: jest.fn()
  }
  renderer.render(
    <TopicSupportComingSoon style={styles} route={route} navigation={navigation} />
  )
  const actual = renderer.getRenderOutput()
  expect(actual).toMatchSnapshot()
})
