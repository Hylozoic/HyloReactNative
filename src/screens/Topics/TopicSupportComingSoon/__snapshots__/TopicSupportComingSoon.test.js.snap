// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches last snapshot 1`] = `
<SafeAreaViewMock>
  <View>
    <Text>
      We're working on expanding #topics to more places.
    </Text>
  </View>
  <View>
    <Text>
      In the meantime, click a #topic from an individual community to see posts from that community.
    </Text>
  </View>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../assets/Axolotyl_Digging.png",
      }
    }
  />
  <View>
    <LinkButton
      to="/"
    >
      <Text>
        Go Back
      </Text>
    </LinkButton>
  </View>
</SafeAreaViewMock>
`;
